src/relay/transforms/to_cps.cc

```diff
@@ -311,15 +313,17 @@ Function ToCPS(const Function& f, const IRModule& m) {
 Function UnCPS(const Function& f) {
   CheckFeature(f, FeatureSet::All() - fGraph);
   ICHECK_GT(f->params.size(), 0);
-  std::vector<Var> new_params;
+  Array<Var> new_params;
   for (const auto& p : f->params) {
-    new_params.push_back(Var(p->name_hint(), p->checked_type()));
+    // TODO(@electriclilies): Not sure if this is correct, it was copying before,
+    // but seems like we just need to make a copy to pop so should be fine?
+    new_params.push_back(WithFields(std::move(p)));
```

Comment:
I'd retain the copies here -- it is building a new function with new parameters.

I think including all args including the span in the ctor will ensure any new fields will at least trigger a compilation error so these few examples of explicit shallow copies can be maintained. Adding a generic ShallowCopy or Clone helper seems overkill.

Reply:
Done, thanks!